pkg/relayer/miner/gen/gen_fixtures.go

```diff
@@ -0,0 +1,334 @@
+// are generated by randomizing the signature of a relay. It generates these mined
+// relay fixtures continuously until the context is canceled. It also returns an
+// error channel which will receive any error it encounters while generating.
+func genRandomizedMinedRelayFixtures(
+	ctx context.Context,
+	randLength int,
+	newHasher func() hash.Hash,
+) (observable.Observable[*relayer.MinedRelay], <-chan error) {
+	var (
+		errCh                      = make(chan error, 1)
+		randBzObs, randBzPublishCh = channel.NewObservable[*relayer.MinedRelay]()
+	)
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			randBz := make([]byte, randLength)
+			if _, err := rand.Read(randBz); err != nil {
+				errCh <- err
+				return
+			}
+
+			// Populate a relay with the minimally sufficient randomized data.
+			relay := servicetypes.Relay{
+				Req: &servicetypes.RelayRequest{
+					Meta: &servicetypes.RelayRequestMetadata{
+						Signature: randBz,
+					},
+					Payload: nil,
+				},
+				Res: nil,
+			}
+
+			// TODO_BLOCKER: use canonical codec.
+			relayBz, err := relay.Marshal()
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			// Hash relay bytes
+			relayHash, err := hashBytes(newHasher, relayBz)
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			randBzPublishCh <- &relayer.MinedRelay{
+				Relay: relay,
+				Bytes: relayBz,
+				Hash:  relayHash,
+			}
+		}
+	}()
+
+	return randBzObs, errCh
+}
+
+// hashBytes hashes the given bytes using the given hasher.
+func hashBytes(newHasher func() hash.Hash, relayBz []byte) ([]byte, error) {
+	hasher := newHasher()
+	if _, err := hasher.Write(relayBz); err != nil {
+		return nil, err
+	}
+
+	return hasher.Sum(nil), nil
+}
+
+// exitOnError exits the program if an error is received on the given error
+// channel.
+func exitOnError(errCh <-chan error) {
+	go func() {
+		for err := range errCh {
+			log.Fatalf("ERROR: %s", err)
+		}
+	}()
+}
+
+// appendMinableRelays maps over the given observable of mined relays, checks
+// their difficulty against flagDifficultyBitsThreshold, and appends them to the
+// given output buffer if it is greater than or equal to the threshold. It stops
+// appending once flagFixtureLimitPerGroup number of relay fixtures have been
+// appended.
+func appendMinableRelays(
+	ctx context.Context,
+	randRelaysObs observable.Observable[*relayer.MinedRelay],
+	outputBuffer *bytes.Buffer,
+) {
+	appendRelays(
+		ctx,
+		randRelaysObs,
+		outputBuffer,
+		func(hash []byte) bool {
+			// Append if difficulty is greater than or equal to threshold.
+			return protocol.MustCountDifficultyBits(hash) >= flagDifficultyBitsThreshold
+		},
+	)
+}
+
+// appendUnminableRelays maps over the given observable of mined relays, checks
+// their difficulty against flagDifficultyBitsThreshold, and appends them to the
+// given output buffer if it is less than the threshold. It stops appending once
```

Comment:
I think so since `flagFixtureLimitPerGroup` is being references directly instead of passed through the function call. `appendRelays` is just a helper to reduce boilerplate.